qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -360,27 +366,30 @@ def _get_evaluate_energy(
 
         def evaluate_energy(parameters: np.ndarray) -> np.ndarray | float:
 
-            try:
-                estimator_job = self.estimator.run(
-                    circuits=[self.ansatz], observables=[operator], parameter_values=[parameters]
-                )
-                estimator_result = estimator_job.result()
-                values = estimator_result.values
-
-            except Exception as exc:
-                raise AlgorithmError("The primitive job to evaluate the energy failed!") from exc
+            estimator_job = self.estimator.run(
+                circuits=[self.ansatz], observables=[operator], parameter_values=[parameters]
+            )
 
+            total_cost = 0
             if step > 1:
-                # Compute overlap cost
+                # compute overlap cost
                 fidelity_job = self.fidelity.run(
                     [self.ansatz] * (step - 1),
                     prev_states,
                     [parameters] * (step - 1),
                 )
+
                 costs = fidelity_job.result().fidelities
 
                 for (state, cost) in zip(range(step - 1), costs):
-                    values += np.real(betas[state] * cost)
+                    total_cost += np.real(betas[state] * cost)
+
+            try:
+                estimator_result = estimator_job.result()
+                values = estimator_result.values + total_cost
```

Comment:
Not blocking but would it be better to take the summation out of the try block to distinguish between the estimator failing and the addition failing (due to e.g. wrong types, or the result being None,...)?

Reply:
sure